### DISCUSSION TITLE
V3 Org UI Feedback

### DISCUSSION THREAD

**Discussion by eriese** [freesewing/freesewing#4257] [Show and tell]
This discussion is meant to serve as a centralized location for discussing improvements to the new v3 ui.

Please post your feedback here, and discuss with others.

If you plan to make a change yourself, please post in this discussion, then make an issue for the change you intend to make and assign it to yourself. This way we can hopefully keep overlapping or conflicting improvements to a minimum.

Thanks, all!

**Comment by bobgeorgethe3rd** [comment]
### Measurements Sets symbol
- Recommend changing it to something like a ruler or Measuring tape as the drafting arm is something that is more relevant to architecture than sewing. It also unfortunately gives the image of forceps.

### Theme, Language & Search
- I feel these 3 should always be on the top menu. When you are in a generated design these are not present and it would be useful to have them if you need to switch language or theme without having to go back a page.

### Measurements
- A good thing to add back would be Manual measurement input like it is in lab 2.5, This will allow people to make a pattern without an account but also will allow users to tweak their measures on the fly when trouble shooting fit issues.
- A Measurements tab on generated design page like lab 2.5 so Measurements can be changed and altered without leaving the generated design page. This would allow again people tweak on the fly and people to change measures without having to navigate backwards.
- I feel the "Curated" label should be removed as this implies more useability and I feel people will think these are good to use which we primairly want people to use custom measures. So it would be "FreeSewing's Measurements Sets". If we do implement more sets from people like we are planning/thinking about doing then the label can be placed back as they are more Curated but atm these are simply 2 sets of measures that are scaled and not graded.
- FreeSewing's Measurements Sets may be best to be a drop down menu and not be the largest part of Measurements page. When dropped down I feel it still needs to tell the user they should use their own measures and that these are really only used to kick the tires so too speak.
- I feel the measurement Categories would be best to be more dynamic and only appear if they are present.
For Example and in order:
  - Bookmarked - appears if there are bookmarked measures
  - Saved - appears if signed-in and have saved measures that have all required measures.
  - Sets with missing measures (appears only if there are some)
  - Manual - If there is saved category then is a dropdown, if there is no saved catergory then appears by default and is not a dropdown. Also partial fill from sets with missing measures could be a good idea.
  - FreeSewing's Measurements Sets - Always there but is dropdown.
- FreeSewing's Measurements Sets are confusingly laid out as they not separated into there sets. What are the tick marks? I have no clue what they are and what they are suppose to represent.

### Designs
- Designs that don't need measures still have the select measurements stage, i.e. Hortensia, Lucy, Madge etc. I assume this is something that will be addressed but I felt it was best to note it.
- Print Layout may be best to have a printer with a magnifying class rather than just a printer like how Print Preview is represented. 
- I feel save and Export maybe should be merged into one. So it's like save to your account and export to print.
- To be honest it might be better to combine Print Layout, Save and Export all into one. This might just be me and I can see the benefits of having them separate as well as together. Maybe could be a control thing mm.
- Reset Zoom button. This is something I was going to try and add for lab 2.5. It's really needed if you have patterns that alter shaped quite a lot like Sandy. Yes you can click on test and then back to designs but this would be a more front-facing solution.

### Test Design
- Clicking twice is adding more steps than I feel are necessary. A single click like before feels more inuitive to me.
- Allow for multiple testing? At least the way it is setup now implies that's a thing and if it's not then changes are needed.

### Design Options
- Missing drop-down arrows. Right now it is not inuitive to the user to click on the option menus and options to tweak them. There is no indication to the user that these can be clicked to allow changes or discover discover the option themselves.
- The bottom of the Categories also having the title and symbols is confusing, I know this is intended to act as a bookend and to close the category from both ends but I feel it implies to click it to drop it down again.
- I feel Reset button should always be presented and not in the bars like it is lab 2.5
- Again I feel Manual option input button should be like in 2.5 and not in the bar.
- To me it just feels clumsy to have them next to the close button and would be more intuitive to be next to slider.
- Missing Bool reset buttons, I don't know if this is missing or just not implemented yet.
- I feel Bools when control is high enough should just be click on and off, the small switch is hard to click and I still found myself clicking on the option producing the drop-down menu rather than using the switch.
- I find text is too small and is hard to read.
- I feel Seam allowance should be on by default. Also should be Include Seam Allowance (I assume the latter is minor).
- Contents does not display to the user in the list which parts have been selected. I'm not sure Only is a better substitute for Contents. Also thinking about it should be Content without the s in this scenario.
- Sub-categories such as advanced.sleevecap, I don't feel are clearer enough to the user on the fact they are a sub-category. I feel the bars should be used instead of the folder icon and should be a clearer colour (the purple in dark mode blends too much)
- Categories being bold like lab 2.5 may also be useful.
- Whilst the symbols are a nice idea I do not feel like they properly convey what the categories are about. I'm not sure what the substitution would be maybe custom symbols.
- I feel like a combination of lab 2.5 and this layout whould be best. lab 2.5 is clearer and more inuitive but extends out too much when going deep into dropdowns. The new UI has boxes which cause less extension which is good but the way these are presented are not inuitive like lab 2.5, combining the style, font and symbols etc of lab 2.5 and combining with the boxes of new UI I feel would be the best bet.

### Menu & Navigation
- I feel the Menu should simply hide and unhide on the left hand side and be like the menu found on the left hand side of the designs page for every page. I feel this should be the only menu. What comes up when you click menu is mismatched to the other menu that is provided and I feel the other one is clearer. 
-When deeper in the left hand side menu should display the page you are so if you click on Aaron, the left Menu will display you as Aaron under designs bit. Whilst I know that is the intention of the  bit at the bottom I feel it should also display this under the catergory. It should not be list of all the patterns as that is confusing. However if you click further into Aaron, say to aaron documentation, it the shows you in docs under Aaron and then if you click on say cutting it displays cutting however in this case it should also display the other main 3 as well for easier navigation.
- I guess I'm suggesting a more curated look of what we have in v2, so it's not just a wall of text and only shows things relevant to where you are if that makes sense. (I am willing to draw this out etc as I know my description here may not be the best)
- Design top pages, top pages that have links to the other aspects of the design like documentation. Also has the Cover art and Showcases.
- When clicking on New and then on Create a new design/plugin it takes you to FreeSewing Dev with no proper distinction between Dev and Org and that you have done so, there is no way back from the Dev to org when you do this. I imagine this will cause confusion for a lot of users. Perhaps a warning that you are leaving FreeSewing.org for FreeSewing.dev like how some sites warn you when you are leaving their site? Also possibly hiding the "Go Further" section when in lower control settings.
- I feel a back button as well as the home button would be useful. Say I am in Aaron but I'm like ooh I want to go back to the generate page, there's not currently a simple one-step way to do this. I feel a back button is more frequently used than a home button and whilst you can simply use your browsers back button I feel it would be better to the user to have that be front-facing.

### Missing Lab functionality with the merger
- If we are merging the site and the lab then I feel features of the lab such as manual measurement input should not be lost during the merger. (I don't know if there are other examples but I feel this needs to be considered) 

### Tags
- I feel tags in designs should be centre aligned like the descriptions and a bit more spaced outfrom each other. Like centre align in columns, so if there are 2 there are 2 columns which each one centre aligned in them.

### Colours
- The purple in design options dark mode needs to be lighter, it is hard to read on the dark background.
- The rainbow shades need to be a bit more differentiated. The greens and the blues are a bit too close together which could cause problems for those who are colour blind. Yes there are symbols but I feel most people read the text. I feel what might be better is to have the background of the top menu be rainbow and lgbt+ colours and have the text symbols be a single colour. This might just be my preference but I feel it's more legible when in one colour but I don't want to lose the inclusivity of the rainbow either.

### No Account Message
Current:
Authentication required

This content is only accessible for FreeSewing members.

This feels kind of gatekeepy but I feel it also implies that there is a pay-wall rather than just and account-wall with the use of "members".

Proposed:
Account required

You need a FreeSewing Account to use this feature.

It's a little more simplistic but feels more on the nose of what you need to do i.e. make an account to get access to these features.

### ChatGPT
- I know this is personal opinion but I feel the website should be chocked full of personal writing to show community. Unless I am mistaken I feel the best way to describe FreeSewing is "A community of users making patterns for users.", if that's not the case or no longer the case than please disregard but I feel the use of AI detracts from that community feel as someone one from the community didn't write it.

**Comment by BenJamesBen** [comment]
(I'm not sure how this Discussion is different than https://github.com/freesewing/freesewing/issues/4071. As I understand it, I was asked to add feedback to that Issue, and now I am being asked to add the same feedback here to this Discussion. In any case, I've copied my feedback below.)

As requested, feedback on the new lab:

- With the browser window at over 75% of my entire screen, the text labels do not appear in the draft navigation top row menu, and it is difficult to see.
- With the browser window at just 50% width of my entire screen, the draft navigation menu disappears entirely. For example, I am unable to get to Edit or Print Layout.
- I don't see a way to clear all settings? No ClearThing equivalent?
- Settings and Measurements seem to clear/reset themselves when I go to a different web page and then back to the lab. But, I think this might be the new, expected behavior?
- 5 levels of Control is nice granularity, but I anticipate all customers simply automatically switching to level 5, regardless of what they actually might want or need. "I don't know what I might want or need, and I don't know which control level will gets me what, so it is safer to just jump to level 5."
- It is inconvenient having the Control setting available only in Draft. I ended up in Printing Layout, looking for a way to change margin size, cover page, etc, but these options were missing (because of my Control level) and there wasn't an obvious way to get them to appear. Another reason to just switch to Level 5.
- There doesn't seem to be a way to edit individual Measurements in the new lab. I miss this feature. (Edit YAML is a workaround.)
- This might just be a familiarity issue, but it is difficult to determine where in the navigation I am currently, just by looking at the navigation bar. It might be helpful to have the Cutting Layout icon dimmed in the navigation bar if I am currently in the Cutting Layout screen. Or, some other way to designate where I am.
- I would recommend another horizontal line in the right menu between print layout options and design options, the same way there is a line between the design options and the core options. To make things easier to see.

(I used the new lab to perform an actual, real-world task, to generate a custom Unice based on my measurements (using Edit YAML to load my measurements) and export the pattern to PDF. I'm actually making a Unice to help test the design.)

Edit: added:

- When Seam Allowance is enabled, the UI says that the default seam allowance that is being applied is 1 cm. However, actually only a 5 mm seam allowance is applied to the pattern. This is confirmed in the YAML. And, selecting the reset button changes the UI seam allowance back to 1 cm, but the color is still magenta indicating a non-default value. (I think that there might be 2 different default values in the code.)
- My perception is that the UI requires too many clicks to perform some actions. To toggle Cutting Layout grain direction, you need to click on Grain Direction to expose the choices and then select Vertical. and then possibly click again to hide the choices. 
- It might just be not yet implemented, but I don't see a place to view log messages.

**Comment by starfetch** [comment]
Also finally got around to taking a look at the new layout. 
First off, it looks very sleek and modern! I like it.
A few things I noticed that could be improved upon:

- Personlly, I have a strong dislike for icons without text (probably stems from having to use a program at work that is the epitomy of mad UI design, and it consists of rows upon rows of unlabelled icons, where you have to get creative in interpreting them, but I digress). 
 I understand it's a space issue, but then could a mouse-over text be added? Not feasible for mobile, but a compromise for PC.
Especially talking about these: 
![image](https://github.com/freesewing/freesewing/assets/80041179/815a1f4b-bf55-44f6-80eb-474428a85d7b)
What do these icons mean? 
- The "test pattern" view of `design options` are identical, which confused me for a second. How about changing the boxes from just "Fit" and "Style" to "Test Fit" and "Test Style"?
- (Not strictly feedback, but I love the placeholder names of "sloganCome. sloganStay")
- The exported pdf now includes `fabric` and `interfacing` pages. I'm not sure what these are supposed to show? Cutting layout? Could there be an option to change what is in the final pdf? (Also might be a point to include instructions, for people that do like to print them off.)
- On the pattern draft, I'm a bit confused about scrolling. I like a seperately scrolling side bar for options, so I don't loose sight of the pattern anymore - great! But while checking this out I somehow zoomed in on the pattern (also great that that works now!), but I can't figure out how to unzoom or zoom further.
- I'm conflicted about having the top-level navigation tripled (on top and on the side and in menu).
- Completely agree with @nataliasayang on the tags.
- I love the fact that changing the language finally lets you stay on the same page you where on before. Yay! Also love the fact that you get the language name in English AND the currently slelected language.
- Teeny tiny things my typography-minded brain cannot let go of:
   - Designs page: The text is not aligned in the centre of the purple bubble: 
![image](https://github.com/freesewing/freesewing/assets/80041179/4f3ba546-efd9-40a8-af46-f3bd7dc1b6f4)
Especially the downstroke of the `p` is almost touching the lower border.
  - Designs page: Having the description of the designs centered gives an inconsiststent look because of the different text length. If you follow @nataliasayang's mockup from Discord for lineart, having the text as ragged right will probably look better.

All in all, great job, @joostdecock! People will be bamboozled when v3 hits the interwebs.

**Comment by joostdecock** [comment]
Closing to increase visibility of active discussions
